src/item.cpp

```diff
@@ -5721,6 +5714,28 @@ float item::get_freeze_point() const
     return get_comestible()->freeze_point;
 }
 
+item *item::get_food()
+{
+    if( is_food() ) {
```

Comment:
You can avoid some code duplication by forwarding to the other getter:
```C++
return const_cast<item*>( const_cast<const item *>( this )->get_food() );
```

Reply:
I could, but I don't think it's worth the `const_cast`.  If the code duplication is deemed sufficiently problematic, I'd instead solve that with a template helper function.

Reply:
Well, this PR is to get rid of code duplication, so it ... adds duplicated code? A helper function template is also a solution.

Reply:
I was on the fence about it, but sure.  Now done.